src/lib/checksum.lua

```diff
@@ -98,7 +154,102 @@ end
+   local s = [[
```

Comment:
AFAIU this change is unrelated to the patch; feel free to submit separately, or not at all if you don't feel like submitting it.